python/ngen_cal/src/ngen/cal/ngen.py

```diff
@@ -74,9 +74,10 @@ class NgenBase(ModelExec):
     # an existing realization to build various calibration realization configs
     # but we should probably take a closer look at this in the near future
     realization: FilePath
-    catchments: FilePath
-    nexus: FilePath
-    crosswalk: FilePath
+    hydrofabric: Optional[FilePath]
```

Comment:
I didn't realize that these were using `pydantic.FilePath`. We really shouldn't be using that anywhere directly. During validation `FilePath` checks if the file exists and that is not always desirable (e.g. validating but don't care, yet, if the files exist). I wrote a [wrapper type](https://github.com/NOAA-OWP/t-route/blob/1a5ccf243db939c186c0cdfe00c080b25cce5ab0/src/troute-config/troute/config/types.py#L15) for `t-route` that kind of gets you the best of both worlds and allows you to turn on a strict mode when you do want to check that files are present during validation. This seems out of scope for this PR though. For now, can you change these to just `Optional[Path]` and leave a note that they should be file paths?

Ill open an issue to switch these over to something like what I introduced in `t-route`.

Reply:
For calibration config, I chose `FilePath` particularly for those semantics, as there wasn't much use case for calibration configs if they weren't being used to execute calibration, which makes these files paths required to exist for that to succeed.  I can be convinced that a use case exists which a calibration config would need to be validated "out of band" in which case adopting your wrapper proposal would help support both cases.

Reply:
> ... there wasn't much use case for calibration configs if they weren't being used to execute calibration ...

At minimum, there is the use case for DMOD to create ngen-cal configurations at calibration job request time from user input (and there probably are others).  Using `FilePath` like this would effectively keep us from doing that without having such things all in the identical paths on whatever was creating the ngen-cal config, or doing some kind of lower-level string replacement on the resulting config file before it was used inside a worker.

Reply:
Sounds reasonable, we will change these in another PR following #131